src/main/java/com/microsoft/azure/functions/worker/broker/FactoryJavaMethodExecutor.java

```diff
@@ -0,0 +1,23 @@
+package com.microsoft.azure.functions.worker.broker;
+
+import com.microsoft.azure.functions.rpc.messages.BindingInfo;
+import com.microsoft.azure.functions.worker.WorkerLogManager;
+import com.microsoft.azure.functions.worker.description.FunctionMethodDescriptor;
+import com.microsoft.azure.functions.worker.reflect.ClassLoaderProvider;
+import org.apache.commons.lang3.SystemUtils;
+
+import java.net.MalformedURLException;
+import java.util.Map;
+
+public class FactoryJavaMethodExecutor {
+    public JavaMethodExecutor getJavaMethodExecutor(FunctionMethodDescriptor descriptor, Map<String, BindingInfo> bindings, ClassLoaderProvider classLoaderProvider)
```

Comment:
I think this would be nice as a static `create(...)` method on the `JavaMethodExecutor` interface (then no need for this Factory class)

Reply:
Thank you so much @trask, Just wondering is it common to have the interface has reference to impl classes? I mean if I moved to the interface I will need to import both impl. So if I delete any impl I need to update the interface.

Reply:
Static methods aren't part of the interface contract, so just a way to organize static helper methods.

I think it makes for a nicer/more discoverable API in a case like this, since you typically know the interface that you want to instantiate, and now you don't need go find/remember the name of a separate factory class to do the instantiation.

One example where I've seen this used to good effect recently: https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/util/context/Context.java

But definitely no worries if you keep it the way it is.

Reply:
@trask thank you so much for your help. Sounds awesome I will take a note to be updated in the next PR.